File: gamewindow.cc
#include "gamewindow.hh"

GameWindow::GameWindow(void)
{
  m_game = new Game;
  ui.setupUi(this);
  QHBoxLayout *layoutHuman = new QHBoxLayout(ui.humanFrame);
  m_humanBoard = new BoardView(m_game->human(), true);
  layoutHuman->addWidget(m_humanBoard);
  QHBoxLayout *layoutComputer = new QHBoxLayout(ui.computerFrame);
  m_computerBoard = new BoardView(m_game->computer(), false);
  layoutComputer->addWidget(m_computerBoard);
  connect(ui.actionNewGame, SIGNAL(triggered()), this, SLOT(restart()));
  connect(ui.actionQuit, SIGNAL(triggered()), this, SLOT(close()));
  connect(ui.startButton, SIGNAL(clicked()), this, SLOT(startGame()));
  connect(m_humanBoard, SIGNAL(message(QString)), this, SLOT(status(QString)));
  connect(m_computerBoard, SIGNAL(computerMove()), this, SLOT(computerMove()));
}

GameWindow::~GameWindow(void)
{
  delete m_game;
}

void GameWindow::restart(void)
{
  delete m_game;
  m_game = new Game;
  m_humanBoard->setBoard(m_game->human());
  m_computerBoard->setBoard(m_game->computer());
  ui.startButton->setEnabled(true);
}

void GameWindow::startGame(void)
{
  m_game->setPlacing(false);
  ui.startButton->setEnabled(false);
};

void GameWindow::computerMove(void)
{
  if (m_game->computer()->defeated()) {
    status("HUMAN WINS!!!");
  } else {
    m_game->computerMove();
    m_humanBoard->update();
    if (m_game->human()->defeated()) {
      status("COMPUTER WINS!!!");
    };
  };
}

void GameWindow::status(QString text)
{
  statusBar()->showMessage(text, DELAY);
}

File: gamewindow.py
from PyQt4.QtGui import QHBoxLayout, QMainWindow
from ui_gamewindow import Ui_GameWindow
from game import Game
from boardview import BoardView
class GameWindow(QMainWindow):
  DELAY = 3000
  def __init__(self):
    super(GameWindow, self).__init__()
    self.game = Game()
    self.ui = Ui_GameWindow()
    self.ui.setupUi(self)
    layoutHuman = QHBoxLayout(self.ui.humanFrame)
    self.humanBoard = BoardView(self.game.human, True)
    layoutHuman.addWidget(self.humanBoard)
    layoutComputer = QHBoxLayout(self.ui.computerFrame)
    self.computerBoard = BoardView(self.game.computer, False)
    layoutComputer.addWidget(self.computerBoard)
    self.ui.actionNewGame.triggered.connect(self.restart)
    self.ui.actionQuit.triggered.connect(self.close)
    self.ui.startButton.clicked.connect(self.startGame)
    self.humanBoard.message.connect(self.status)
    self.computerBoard.computerMove.connect(self.computerMove)
  def restart(self):
    self.game = Game()
    self.humanBoard.setBoard(self.game.getHuman())
    self.computerBoard.setBoard(self.game.getComputer())
    self.ui.startButton.setEnabled(True)
  def startGame(self):
    self.game.setPlacing(False)
    self.ui.startButton.setEnabled(False)
  def computerMove(self):
    if self.game.computerDefeated():
      self.status("HUMAN WINS!!!")
    else:
      self.game.computerMove()
      self.humanBoard.update()
      if self.game.humanDefeated():
        self.status("COMPUTER WINS!!!")
  def status(self, text):
    self.statusBar().showMessage(text, self.DELAY)

